### PR TITLE
Security Advisory #GHSA-g6qj-pj38-2465: Address CVE-2024-21634

### PR DESCRIPTION
**Pull request by ALRubinger**
Addresses https://github.com/TBD54566975/web5-kt/security/advisories/GHSA-g6qj-pj38-2465

### CODE REVIEW

**Comment by ALRubinger** [comment]
image
This passes FOSSA where it was failing before, newly discovered 3 days ago. Resolves the issue. This forced upgrade to `com.amazonaws:aws-java-sdk-kms:1.12.668` strips out the problematic `software.amazon.ion:ion-java` dependency entirely. TestSuite passes so I assume all is OK?

@jiyoontbd @frankhinek, others, mind an approval and merge here?

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -24,7 +24,7 @@
 ###
 
 [versions]
-com_amazonaws_aws_kms = "1.12.538"
+com_amazonaws_aws_kms = "1.12.668" # Addresses https://github.com/TBD54566975/web5-kt/security/advisories/GHSA-g6qj-pj38-2465
```
Comment (jiyoontbd) [inline gradle/libs.versions.toml:27]: i don't think we need a comment here coz this PR will be merged into main and the PR title which becomes the commit message describes why this version was bumped

**Review by jiyoontbd — APPROVED** [review]

File: gradle/libs.versions.toml (hunk 1 above)
Comment (frankhinek) [inline gradle/libs.versions.toml:27]:
```suggestion
com_amazonaws_aws_kms = "1.12.668"
```

**Review by frankhinek — CHANGES_REQUESTED** [review]

File: gradle/libs.versions.toml (hunk 1 above)
Comment (ALRubinger) [reply gradle/libs.versions.toml:27]: Why remove the doc to future maintainers about why this version, @frankhinek?

Comment (jiyoontbd) [reply gradle/libs.versions.toml:27]: i think it may be because we might change this version again and then the comment can easily get left "orphaned"?

Comment (ALRubinger) [reply gradle/libs.versions.toml]: Updated PR to accept @frankhinek suggestion, rebased

**Review by tomdaffurn — APPROVED** [review]

**Review by frankhinek — APPROVED** [review]

**Pull request merged**